crates/sui-indexer/src/store/pg_indexer_store_v2.rs

```diff
@@ -74,6 +74,35 @@ const PG_COMMIT_PARALLEL_CHUNK_SIZE_PER_DB_TX: usize = 500;
 // Having this number too high may cause many db deadlocks because of
 // optimistic locking.
 const PG_COMMIT_OBJECTS_PARALLEL_CHUNK_SIZE_PER_DB_TX: usize = 500;
+const OBJECTS_SNAPSHOT_MAX_CHECKPOINT_LAG: usize = 900;
+const OBJECTS_SNAPSHOT_MIN_CHECKPOINT_LAG: usize = 300;
+
+const UPDATE_OBJECTS_SNAPSHOT_QUERY: &str = r"
+INSERT INTO objects_snapshot (object_id, object_version, object_status, object_digest, checkpoint_sequence_number, owner_type, owner_id, object_type, serialized_object, coin_type, coin_balance, df_kind, df_name, df_object_type, df_object_id)
+SELECT object_id, object_version, object_status, object_digest, checkpoint_sequence_number, owner_type, owner_id, object_type, serialized_object, coin_type, coin_balance, df_kind, df_name, df_object_type, df_object_id
+FROM (
+    SELECT *,
+           ROW_NUMBER() OVER (PARTITION BY object_id ORDER BY object_version DESC) as rn
+    FROM objects_history
+    WHERE checkpoint_sequence_number >= $1 AND checkpoint_sequence_number < $2
+) as subquery
+WHERE rn = 1
```

Comment:
Is this a fancy way of doing "get all objects with their greatest versions"? Having a comment explaining here would be great. Thanks!

Reply:
yes, `rn` is a generated rank from ROW_NUMBER(), which is ordered rank based on desc object version.

Reply:
added the comments.